src/components/MAPI/clusters/Clusters.tsx

```diff
@@ -95,7 +95,7 @@ const Clusters: React.FC<{}> = () => {
     clusterListIsValidating;
 
   const sortedClusters = useMemo(
-    () => clusterList?.items.sort(compareClusters),
+    () => clusterList?.items.slice().sort(compareClusters),
```

Comment:
After deleting a cluster and mutating the cache, the newly created array is a `ReadonlyArray`, which means that we need to copy it before sorting, so it won't throw any errors.